### PR TITLE
add support for MultiSelectListPreference

### PR DESCRIPTION
**Pull request by AriaMoradi**

### CODE REVIEW

**Comment by AriaMoradi** [comment]
@Syer10 Your implementation of `StringSet` for `JavaSharedPreferences` didn't work, so I rewrote it using Json, if a better/fixed implementation is existed then that would be desirable.

File: AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt [hunk 1]
```diff
@@ -143,7 +144,10 @@ class JavaSharedPreferences(key: String) : SharedPreferences {
             itemsToAdd.forEach { (key, value) ->
                 @Suppress("UNCHECKED_CAST")
                 when (value) {
-                    is Set<*> -> preferences.encodeValue(SetSerializer(String.serializer()), key, value as Set<String>)
+                    is Set<*> -> preferences.putString(
+                        key,
+                        json.encodeToString((value as Set<String>).toList())
```
Comment (Syer10) [inline AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt]: KotlinX supports encoding/decoding Set<>
```suggestion
                        json.encodeToString(value as Set<String>)
```

File: AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt [hunk 2]
```diff
@@ -27,6 +24,10 @@ class JavaSharedPreferences(key: String) : SharedPreferences {
     private val preferences = JvmPreferencesSettings(javaPreferences)
     private val listeners = mutableMapOf<SharedPreferences.OnSharedPreferenceChangeListener, PreferenceChangeListener>()
 
+    companion object {
+        private val json = Json { ignoreUnknownKeys }
```
Comment (Syer10) [inline AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt]:
```suggestion
        private val json = Json { ignoreUnknownKeys = true }
```

File: AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt [hunk 3]
```diff
@@ -40,15 +41,15 @@ class JavaSharedPreferences(key: String) : SharedPreferences {
...
-            return if (defValues != null) {
-                preferences.decodeValue(SetSerializer(String.serializer()).nullable, key, defValues)
-            } else {
-                preferences.decodeValue(SetSerializer(String.serializer()).nullable, key, null)
-            }?.toMutableSet()
-        } catch (e: SerializationException) {
-            throw ClassCastException("$key was not a StringSet")
+    override fun getStringSet(key: String, defValues: MutableSet<String>?): Set<String>? {
+        val value = if (defValues != null) {
+            preferences.getString(key, json.encodeToString(defValues.toList()))
```
Comment (Syer10) [inline AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt]:
```suggestion
            preferences.getString(key, json.encodeToString(defValues))
```

File: AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt [hunk 4]
```diff
@@ -40,15 +41,15 @@ class JavaSharedPreferences(key: String) : SharedPreferences {
...
+        } else {
+            preferences.getStringOrNull(key)
+        }
+
+        return value?.let {
+            json.decodeFromString<List<String>>(it).toSet()
```
Comment (Syer10) [inline AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt]:
```suggestion
            json.decodeFromString<Set<String>>(it)
```

**Review by Syer10 — CHANGES_REQUESTED** [review]

**Comment by Syer10** [comment]
> @Syer10 Your implementation of `StringSet` for `JavaSharedPreferences` didn't work, so I rewrote it using Json, if a better/fixed implementation is existed then that would be desirable.

Thats really weird, thats the standard way for handling sets/list with Multiplatform Preferences

**Review by Syer10 — APPROVED** [review]

**Pull request merged**